src/olympia/accounts/views.py

```diff
@@ -774,3 +778,88 @@ def post(self, request):
                 _('Notification [%s] does not exist') % notification_name
             )
         return Response(serializer.data)
+
+
+class FxaNotificationView(FxAConfigMixin, APIView):
+    authentication_classes = []  # TODO: reimplement FxA auth in authentication_class?
+    permission_classes = []
+
+    FXA_PROFILE_CHANGE_EVENT = (
+        'https://schemas.accounts.firefox.com/event/profile-change'
+    )
+    FXA_DELETE_EVENT = 'https://schemas.accounts.firefox.com/event/delete-user'
+    FXA_PASSWORDCHANGE_EVENT = (
+        'https://schemas.accounts.firefox.com/event/password-change'
+    )
+
+    def get_fxa_verifying_keys(self):
+        if not hasattr(self, 'fxa_verifying_keys'):
+            response = requests.get(f'{settings.FXA_OAUTH_HOST}/jwks')
+            self.fxa_verifying_keys = (
+                response.json().get('keys') if response.status_code == 200 else []
+            )
+
+        if not self.fxa_verifying_keys:
+            raise exceptions.AuthenticationFailed(
+                'FXA verifying keys are not available.'
+            )
+
+        return self.fxa_verifying_keys
+
+    def get_jwt(self, request):
+        request_jwt = request.headers['Authorization'].split('Bearer ')[1]
+        client_id = self.get_fxa_config(request)['client_id']
+
+        for verifying_key in self.get_fxa_verifying_keys():
+            if verifying_key['alg'] != 'RS256':
+                # we only support RS256
+                continue
+            algorithm = jwt.algorithms.RSAAlgorithm.from_jwk(verifying_key)
+            authenticated_jwt = jwt.decode(
+                request_jwt,
+                algorithm,
+                audience=client_id,
+                algorithms=[verifying_key['alg']],
+            )
+
+        if not authenticated_jwt:
+            raise exceptions.AuthenticationFailed(
+                'Could not authenticate JWT with FXA key.'
+            )
+
+        return authenticated_jwt
+
+    def process_event(self, uid, event_key, event_data):
+        timestamp = event_data.get('changeTime') or datetime.now().timestamp()
+        if event_key == self.FXA_PROFILE_CHANGE_EVENT:
+            log.info(f'Fxa Webhook: Got profile change event for {uid}')
+            new_email = event_data.get('email')
+            if not new_email:
+                log.info(
+                    'Email property missing/empty for "%s" event; ignoring' % event_key
+                )
+            else:
+                primary_email_change_event.delay(uid, timestamp, new_email)
+        if event_key == self.FXA_DELETE_EVENT:
+            log.info(f'Fxa Webhook: Got delete event for {uid}')
+            delete_user_event.delay(uid, timestamp)
+        if event_key == self.FXA_PASSWORDCHANGE_EVENT:
+            log.info(f'Fxa Webhook: Got password-change event for {uid}')
+            clear_sessions_event.delay(uid, timestamp, 'password-change')
+
+    def post(self, request):
+        jwt = self.get_jwt(request)
```

Comment:
I would call this `payload` rather than `jwt` to avoid confusion with the `jwt` module (and to be consistent with how we call this in our existing auth code)